docs/cryptodoc/src/05_06_hss_lms.rst

```diff
@@ -0,0 +1,416 @@
+   | LMOTS_SHA256_N24_W2 | SHA-256/192  | 24  | 2   | 101 | 6   | 0x0006 |
+   +---------------------+--------------+-----+-----+-----+-----+--------+
+   | LMOTS_SHA256_N24_W4 | SHA-256/192  | 24  | 4   | 51  | 4   | 0x0007 |
+   +---------------------+--------------+-----+-----+-----+-----+--------+
+   | LMOTS_SHA256_N24_W8 | SHA-256/192  | 24  | 8   | 26  | 0   | 0x0008 |
+   +---------------------+--------------+-----+-----+-----+-----+--------+
+   | LMOTS_SHAKE_N32_W1  | SHAKE256/256 | 32  | 1   | 265 | 7   | 0x0009 |
+   +---------------------+--------------+-----+-----+-----+-----+--------+
+   | LMOTS_SHAKE_N32_W2  | SHAKE256/256 | 32  | 2   | 133 | 6   | 0x000a |
+   +---------------------+--------------+-----+-----+-----+-----+--------+
+   | LMOTS_SHAKE_N32_W4  | SHAKE256/256 | 32  | 4   | 67  | 4   | 0x000b |
+   +---------------------+--------------+-----+-----+-----+-----+--------+
+   | LMOTS_SHAKE_N32_W8  | SHAKE256/256 | 32  | 8   | 34  | 0   | 0x000c |
+   +---------------------+--------------+-----+-----+-----+-----+--------+
+   | LMOTS_SHAKE_N24_W1  | SHAKE256/192 | 24  | 1   | 200 | 8   | 0x000d |
+   +---------------------+--------------+-----+-----+-----+-----+--------+
+   | LMOTS_SHAKE_N24_W2  | SHAKE256/192 | 24  | 2   | 101 | 6   | 0x000e |
+   +---------------------+--------------+-----+-----+-----+-----+--------+
+   | LMOTS_SHAKE_N24_W4  | SHAKE256/192 | 24  | 4   | 51  | 4   | 0x000f |
+   +---------------------+--------------+-----+-----+-----+-----+--------+
+   | LMOTS_SHAKE_N24_W8  | SHAKE256/192 | 24  | 8   | 26  | 0   | 0x0010 |
+   +---------------------+--------------+-----+-----+-----+-----+--------+
+
+In addition to these parameters, an LM-OTS
+instance is defined by the identifier ``I`` of the LMS tree
+and the index of its leaf ``q``, where the LM-OTS instance is located; this is
+represented by the class ``OTS_Instance``.
+For each LM-OTS instance, we can create a keypair with a secret key (class
+``LMOTS_Private_Key``) and a public key (class ``LMOTS_Public_Key``). The
+pseudorandom key generation method recommended in [RFC8554]_ Appendix A derives the
+secret key's Winternitz chain inputs (``x[]`` of [RFC8554]_). The inputs for this
+method are the LM-OTS instance parameters and a
+secret seed ``SEED`` associated with an LMS tree:
+
+.. math::
+   \mathtt{x[i]\ =\ Hash(I\ ||\ u32str(q)\ ||\ u16str(i)\ ||\ u8str(0xff)\ ||\ SEED)}
+
+The public key is created by computing all Winternitz hash chains
+beginning with their secret chain inputs ``x[]`` (see [RFC8554]_ Algorithm 1).
+Besides the instance parameters, it contains the final hash value denoted as
+``K`` in [RFC8554]_ Algorithm 1.
+
+For creating an LM-OTS signature of a message, Botan offers the method
+``LMOTS_Private_Key::sign``. For that, it implements Algorithm 1 of [RFC8554]_.
+One important remark is the creation of the randomizer ``C``. To create this
+randomizer, Botan adapts the same approach as the reference implementation by
```

Comment:
Well, the implementation is mentioned in [RFC 8554 Appendix E](https://datatracker.ietf.org/doc/html/rfc8554#appendix-E), so I think it can be called somehow official. I can mention Cisco, though.

Reply:
Missed that it is mentioned in the Appendix, I thought it was only referenced for the measurements taken.